src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -25,7 +24,19 @@ private static partial class Settings
         // This allows for the whole Settings nested class to be trimmed when Invariant=true, and allows for the Settings
         // static cctor (on Unix) to be preserved when Invariant=false.
         internal static bool Invariant => Settings.Invariant;
-#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS || TARGET_BROWSER
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+        private static bool GetHybridGlobalizationSwitchValue() {
+            if (Settings.Invariant)
+                return false;
+            string? value;
+            TryGetStringValue("System.Globalization.Hybrid", "DOTNET_SYSTEM_GLOBALIZATION_HYBRID", out value);
+            if (value == null || value == "1")
+                return true;
+            return false;
+        }
+
+        internal static bool Hybrid => GetHybridGlobalizationSwitchValue();
```

Comment:
```suggestion
        internal static bool Hybrid => Settings.Invariant ? false : AppContextConfigHelper.GetBooleanConfig("System.Globalization.Hybrid", "DOTNET_SYSTEM_GLOBALIZATION_HYBRID", true);
```

Is `DOTNET_SYSTEM_GLOBALIZATION_HYBRID` ever true when `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` is true? For wasm it never happens, we take care of it when setting these config variables. That's why we do not check for `Settings.Invariant` when setting `Settings.Hybrid`. The separation from `TARGET_BROWSER`-specific code might be unnecessary when you unify the behavior and make sure multiple config variables cannot be set to true at the same time.

Reply:
Refactored this part. There is a check in Xcode.cs already for Invariant/Hybrid, no need to do it here for Apple also